storage/innobase/lock/lock0lock.cc

```diff
@@ -6428,3 +6443,86 @@ void lock_update_split_and_merge(
                           PAGE_HEAP_NO_SUPREMUM,
                           lock_get_min_heap_no(right_block));
 }
+#ifdef WITH_WSREP
+/*
+  Wsrep BF watchdog implementation:
+
+  We use InnoDB monitor thread to call wsrep_run_BF_lock_wait_watchdog()
+  periodically to check if there are applier threads hanging in
+  lock wait. If the lock wait has taken more than 5 seconds, the
+  potential local blocking transactions are killed until the
+  applier thread is released.
+ */
+
+std::atomic<uint> wsrep_BF_waiting_count;
+uint innodb_wsrep_applier_lock_wait_timeout =
+    INNODB_WSREP_APPLIER_LOCK_WAIT_TIMEOUT_DEFAULT;
+
+struct wsrep_BF_lock_waiters_t
+{
+  std::vector<trx_t *, ut_allocator<trx_t *> > bf_waiters{};
+  ulonglong max_wait_time{0};
+  my_hrtime_t now{my_hrtime_coarse()};
```

Comment:
A non-constant initialization expression for an implicit default constructor is not allowed in the oldest compilers that we support (GCC 4.8.5):
```
/home/buildbot/amd64-centos-7-rpm-autobake/build/padding_for_CPACK_RPM_BUILD_SOURCE_DIRS_PREFIX/storage/innobase/lock/lock0lock.cc:6505:37: error: cannot convert 'my_hrtime_t' to 'ulonglong {aka long long unsigned int}' in initialization
   my_hrtime_t now{my_hrtime_coarse()};
```